### PR TITLE
Add support for NPM 8, 9, 10 and PNPM 8, remove support for NPM 6

### LINKED ISSUE #88: Javascript setup is not compatible with npm 7 / 8 / 9
The javascript setup in the `assets/admin` folder uses the `file:...` syntax to require Sulu packages such as the `sulu-admin-bundle`:
https://github.com/sulu/skeleton/blob/c9640ee7dae09f09a84d83fce0b5143d52fc4a84/assets/admin/package.json#L15-L28

Dependencies that are required like this will be symlinked into the `assets/admin/node_moules` directory. Unfortunately, `npm v7` does not install the dependencies of these dependencies at the moment: https://github.com/npm/cli/issues/2339 

Because of this, the dependencies of the Sulu bundles are not installed and the build fails with a lot of error messages like this:

```
ERROR in /sulu-skeleton/vendor/sulu/sulu/src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs/Tabs.js
Module not found: Error: Can't resolve 'classnames' in '/sulu-skeleton/vendor/sulu/sulu/src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs'
 @ /sulu-skeleton/vendor/sulu/sulu/src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs/Tabs.js 54:0-36 347:22-32
 @ /sulu-skeleton/vendor/sulu/sulu/src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs/index.js
 @ /sulu-skeleton/vendor/sulu/sulu/src/Sulu/Bundle/AdminBundle/Resources/js/components/index.js
 @ /sulu-skeleton/vendor/sulu/sulu/src/Sulu/Bundle/ContactBundle/Resources/js/components/ContactDetails/Phone.js
 @ /sulu-skeleton/vendor/sulu/sulu/src/Sulu/Bundle/ContactBundle/Resources/js/index.js
 @ ./index.js
```

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | yes
| Deprecations? | no
| Fixed tickets | fixes #88 
| Related issues/PRs | #issuenum
| License | MIT
| Documentation PR | sulu/sulu-docs#prnum

#### What's in this PR?

Add support for NPM 8, 9, 10 and PNPM 8, remove support for NPM 6

#### Why?

Node 16 is going for end of life: https://nodejs.org/en/about/previous-releases

#### Example Usage

```php
rm -rf ./**/node_modules/
rm -rf ./**/package-lock.json

cd assets/admin
npm -g i npm@10

npm install
npm run build
```

```php
rm -rf ./**/node_modules/
rm -rf ./**/package-lock.json
rm -rf ./**/pnpm-lock.yaml

cd assets/admin
curl -fsSL https://get.pnpm.io/install.sh | sh -

pnpm install
pnpm run build
```

#### BC Breaks/Deprecations

No support for NPM 6 / symlinked local packages.

#### Disadvantages

No watch task over vendor package developments as install requires `install-links=true` which ends not longer uses symlinks. Instead a tarball is created and install, but it installs correctly dependencies of local dependencies. Which does not work for `file:` packages.

Why this opens a wider range for support of modern npm and pnpm versions. `Yarn` is still a problem and not supported by sulu. This includes not only yarn 1 but also 2,3,4.

### CODE REVIEW

File: assets/admin/.npmrc [hunk 1]
```diff
@@ -13,3 +13,12 @@ engine-strict=true
...
 legacy-peer-deps=true
+
+# unfortunataly, npm >= 7 ignores dependencies of local dependencies to handle them the same way as normal
+# dependencies we require to use install-links which will create a tar ball for all packages and install
+# it like a normal dependency with its dependencies
+install-links=true
+
+# pnpm does not install dependencies of local dependencies correctly by default but if use the shamefully-hoist
+# configuration it also installs them correctly and can so be correctly build via pnpm
+shamefully-hoist=true
```
Comment (alexander-schranz) [inline assets/admin/.npmrc:24]: This are the main changes required for newer NPM version and for using PNPM

**Comment by TheCadien** [comment]
@danielaKay or maybe you could help here :) ?

**Review by wachterjohannes — APPROVED** [review]

**Review by Prokyonn — APPROVED** [review]

**Comment by chirimoya** [comment]
@alexander-schranz how will development work change with the missing watch task? Always reinstall and build? Can that be done with some kind of automation?

**Comment by alexander-schranz** [comment]
@chirimoya 

> how will development work change with the missing watch task? Always reinstall and build? Can that be done with some kind of automation?

Developing in core would be easier for most when just work in the `sulu/sulu` itself not longer over the `sulu/skeleton`.

Developing of third party bundles is more an issue, normally `npm link` would solve it but it ignores the `install-links` flag which make it not usable for sulu. Still manually symlinks would allow still have the same behaviour as today, aslong as there are no conflicting dependencies:

```sh
npm install # install all dependencies

rm -rf node_modules/sulu-admin-bundle
ln -s @sulu/vendor/sulu/sulu/src/Sulu/Bundle/AdminBundle/Resources/js node_modules/sulu-admin-bundle
```

We could provide a script doing this mechanism something like `npm run prepare-dev-links`.

**Review by chirimoya — APPROVED** [review]

File: .github/workflows/test-application.yaml [hunk 2]
```diff
@@ -188,11 +189,11 @@ jobs:
...
                 include:
-                    - php-version: '8.1'
-                      node-version: '16'
-                      npm-version: '6'
+                    - php-version: '8.3'
+                      node-version: '20'
+                      npm-version: '10'
                       mysql-version: '8.0'
-                      php-extensions: 'ctype, iconv, intl, mysql, pdo_mysql, php_fileinfo, imagick'
+                      php-extensions: 'ctype, iconv, intl, mysql, pdo_mysql, php_fileinfo, gd, sodium'
```
Comment (alexander-schranz) [inline .github/workflows/test-application.yaml:196]: Bildschirmfoto 2024-03-15 um 10 31 41

Looks like Imagick on PHP 8.3 is not yet supported for windows.

The `sodium` PHP Extension seems not longer be part of setup-php of winodws. So we need to enable it.

**Pull request merged**